packages/@aws-cdk/core/lib/deps.ts

```diff
@@ -36,20 +79,33 @@ export function addDependency<T extends Element>(source: T, target: T, reason?:
   const targetStage = Stage.of(targetStack);
   if (sourceStage !== targetStage) {
     // eslint-disable-next-line max-len
-    throw new Error(`You cannot add a dependency from '${source.node.path}' (in ${describeStage(sourceStage)}) to '${target.node.path}' (in ${describeStage(targetStage)}): dependency cannot cross stage boundaries`);
+    throw new Error(`You cannot have a dependency from '${source.node.path}' (in ${describeStage(sourceStage)}) to '${target.node.path}' (in ${describeStage(targetStage)}): dependency cannot cross stage boundaries`);
```

Comment:
can we leave the original error message?

Reply:
We can, though when this method is used in the `remove` case, the original message may be a bit confusing.